src/shared/KubectlDetector.cs

```diff
@@ -47,10 +47,16 @@ public static Task<bool> IsKubectlConnectedToClusterAsync(OutputContext output)
                     {
                         continue;
                     }
-                    var major = int.Parse(element.Value.GetProperty("major").GetString());
-                    var minor = int.Parse(element.Value.GetProperty("minor").GetString().Trim('+'));
-                    var version = new Version(major, minor);
-                    return version;
+                    var majorString = element.Value.GetProperty("major").GetString();
+                    var minorString = element.Value.GetProperty("minor").GetString();
+
+                    if (majorString != null && minorString != null)
```

Comment:
will `element.Value.GetProperty("major").GetString()` ever return null?

Reply:
`GetProperty()` will return `null` if the property doesn't exist, which is unlikely, but could conceivably happen.

Reply:
I think it throws error instead of returning null. Even if it returns null, the .GetString() will result into null ref exception.

Reply:
Ah, yes, sorry, my mistake.  It's `GetString()` which could return null (if the property exists but is set to `null`).